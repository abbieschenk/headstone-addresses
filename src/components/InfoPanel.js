import React from "react";

import "../styles/InfoPanel.scss";

const InfoPanel = ({data}) => {
    return (
        <div className="info-panel">
            {data ? 
                data.City ? (
                    <div className="info headstone">
                        <div className="section">
                            <div>
                                <strong>{data.FullNameChinese}</strong> {data.FirstNameEnglish} {data.LastNameEnglish}
                            </div>
                            <div>
                                {data.BirthYear}{data.BirthYear ? "â€“" : "Buried "}{data.DeathYear}    
                            </div>
                        </div>
                        <div className="section">
                            <div>
                                <strong>Address</strong> 
                            </div>
                            <div>
                                {data.HeadstoneNotes}
                            </div>
                            <div>
                                {data.Province}{data.City && ", "}{data.City}{data.Town && ", "}{data.Town}{data.Village && ", "}{data.Village}{data.Neighbourhood && ", "}{data.Neighbourhood}
                            </div>
                        </div>
                        <div className="section">
                            <div>
                                <strong>Cemetery</strong> 
                            </div>
                                {data.CemeteryName} Cemetery
                            <div>
                                {data.Section} {data.Block} {data.Plot}
                            </div>
                        </div>
                        {data.Note && (
                        <div className="section">
                            <i>{data.Note}</i>
                        </div>
                        )}
                    </div>
                ) : ( 
                    <div className="info address">
                        <div>
                            <strong>{data.NameChinese}</strong> {data.NameEnglish}
                        </div>
                    </div>
            ) : (
                <div className="info empty" />
            )}
    
    
        </div>
    );
};

export default InfoPanel;
