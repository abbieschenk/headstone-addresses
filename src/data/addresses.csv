NameEnglish,NameChinese,LocX,LocY,Color
Gat Lam,吉林,145,10,#5AA69C
Soeng Hoi,上海,102,125,#51615F
Gaau Gong,九江,74,130,#946E6D
Naam Ping,南平,90,154,#50B1B1
Gwong Dung,廣東,62,168,#354745
Dung Gun,東莞,70,176,#75947E
Naam Hoi,南海,54,176,#6EC4D3
Pun Jyu,番禺,60,185,#F2AC61
Zhung Saan,中山,64,194,#47352E
Hok Saan,鶴山,51,190,#949475
San Wui,新會,55,198,#94796D
Hoi Ping,開平,46,198,#B58048
Toi Saan,台山,50,206,#6D948F
Jan Ping,恩平,41,206,#806D94
Seoi Kai,遂溪,24,213,#848D94
Hoi Laam,海南,19,234,#E0A7A6