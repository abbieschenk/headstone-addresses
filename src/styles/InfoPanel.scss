@use "./_colors";

#info-panel {
    width: 100%;
    height: 250px;
    padding: 15px;
    
    border: solid 1px colors.$green;
    border-left: none;

    background-color: colors.$background-lighter;

    .section {
        padding-bottom: 15px;
    }
}