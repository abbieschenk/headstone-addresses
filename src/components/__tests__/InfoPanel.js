import React from "react";
import renderer from "react-test-renderer";

import InfoPanel from "../InfoPanel";

import data from './data.json';

describe("InfoPanel", () => {
    const { headstones, addresses } = data;

    const headstoneData = {
        headstone: headstones[0]
    };

    const addressData = {
        address: addresses[0]
    };

    const emptyData = {};

    it("renders headstone data correctly", () => {
        const tree = renderer
            .create(<InfoPanel data={headstoneData} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("renders address data correctly", () => {
        const tree = renderer
            .create(<InfoPanel data={addressData} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("renders empty data correctly", () => {
        const tree = renderer
            .create(<InfoPanel data={emptyData} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});