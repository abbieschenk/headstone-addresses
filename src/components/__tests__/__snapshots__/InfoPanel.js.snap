// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel renders address data correctly 1`] = `
<div
  className="info-panel"
>
  <div
    className="info address"
  >
    <div>
      <strong>
        測試地點-1
      </strong>
       
      Test-City-1
    </div>
  </div>
</div>
`;

exports[`InfoPanel renders empty data correctly 1`] = `
<div
  className="info-panel"
>
  <div
    className="info empty"
  >
    <div
      className="section"
    >
      Chinese headstones often document ancestral hometown addresses. This visualization links headstones to their addresses.
    </div>
    <div
      className="section"
    >
      Select a headstone on the right or hometown on the left to see their links. Zoom by scrolling and pan by dragging.
    </div>
    <div
      className="section"
    >
      Drag the slider to filter by birth date and burial date. Headstones without birth dates will be hidden.
    </div>
    <div
      className="section"
    >
      Best viewed on a larger screen. See 
      <a
        href="/about"
      >
        About
      </a>
       for more information.
    </div>
  </div>
</div>
`;

exports[`InfoPanel renders headstone data correctly 1`] = `
<div
  className="info-panel"
>
  <div
    className="info headstone"
  >
    <div
      className="section"
    >
      <div>
        <strong>
          測試名稱-1
        </strong>
         
        Test-FirstName-1
         
        Test-LastName-1
      </div>
      <div>
        1900
        –
        1980
      </div>
    </div>
    <div
      className="section"
    >
      <div>
        <strong>
          Address
        </strong>
      </div>
      <div>
        Note
      </div>
      <div>
        Test-Province, 
        Test-City-1
        , Test-Town
        , Test-Village
        , Test-Neighbourhood
      </div>
    </div>
    <div
      className="section"
    >
      <div>
        <strong>
          Cemetery
        </strong>
      </div>
      Test-Cemetery
       Cemetery
      <div>
        Test-Section
         
        Test-Block
         
        Test-Plot
      </div>
    </div>
    <div
      className="section"
    >
      <i>
        Test-Note
      </i>
    </div>
  </div>
</div>
`;
